test/forge/TestCaseParser.sol

```diff
@@ -0,0 +1,341 @@
+                "testCaseTemplateByTargetAmount"
+            );
+        } else {
+            (strategies, tradeActions, sourceAmount, targetAmount) = parseTestDataTemplate(
+                "testCaseTemplateBySourceAmount"
+            );
+        }
+
+        if (inverseOrders) {
+            // swap orders for some strategies if true
+            for (uint256 i = 0; i < strategies.length; ++i) {
+                if (i % 2 == 0) {
+                    Order memory tempOrder = strategies[i].orders[0];
+                    Order memory tempOrderExpected = strategies[i].expectedOrders[0];
+
+                    strategies[i].orders[0] = strategies[i].orders[1];
+                    strategies[i].expectedOrders[0] = strategies[i].expectedOrders[1];
+                    strategies[i].orders[1] = tempOrder;
+                    strategies[i].expectedOrders[1] = tempOrderExpected;
+                }
+            }
+        }
+
+        return
+            TestCase({
+                sourceSymbol: sourceSymbol,
+                targetSymbol: targetSymbol,
+                strategies: strategies,
+                tradeActions: tradeActions,
+                byTargetAmount: byTargetAmount,
+                sourceAmount: sourceAmount,
+                targetAmount: targetAmount
+            });
+    }
+
+    /**
+     * @dev helper function to get a test case data struct by parsing test data json (overriden with additional args)
+     */
+    function getTestCase(
+        string memory sourceSymbol,
+        string memory targetSymbol,
+        bool byTargetAmount,
+        bool equalHighestAndMarginalRate,
+        bool inverseOrders
+    ) public returns (TestCase memory testCase) {
+        TestStrategy[] memory strategies;
+        TradeAction[] memory tradeActions;
+        uint256 sourceAmount;
+        uint256 targetAmount;
+        if (equalHighestAndMarginalRate) {
+            if (byTargetAmount) {
+                (strategies, tradeActions, sourceAmount, targetAmount) = parseTestDataTemplate(
+                    "testCaseTemplateByTargetAmountEqualHighestAndMarginalRate"
+                );
+            } else {
+                (strategies, tradeActions, sourceAmount, targetAmount) = parseTestDataTemplate(
+                    "testCaseTemplateBySourceAmountEqualHighestAndMarginalRate"
+                );
+            }
+        } else {
+            if (byTargetAmount) {
+                (strategies, tradeActions, sourceAmount, targetAmount) = parseTestDataTemplate(
+                    "testCaseTemplateByTargetAmount"
+                );
+            } else {
+                (strategies, tradeActions, sourceAmount, targetAmount) = parseTestDataTemplate(
+                    "testCaseTemplateBySourceAmount"
+                );
+            }
+        }
+        if (inverseOrders) {
+            // swap orders for some strategies if true
+            for (uint256 i = 0; i < strategies.length; ++i) {
+                if (i % 2 == 0) {
```

Comment:
You can just use `i += 2` in the `for` loop (then get rid of this check).

Reply:
Implemented